### PR TITLE
Convert Host Overview page to use APi endpoint

### PR DESCRIPTION
**Pull request by mehulkar**
This begins to address #250 by getting the page to interact with an API endpoint. 

I couldn't test that it shows the right info when hosts have teatimes, because I can't create teatimes because of #316 :( I can manually insert db rows later to test. 

Also, this doesn't actually address the different data @ankitshah811 wants in the table (detailed in #250). It's just a first step at converting it to API + client side js. 

Additionally, this fixes one outstanding bug in the app which is that it uses haml :stuck_out_tongue: . 

Isn't this beautiful code?

### CODE REVIEW

File: app/controllers/api/v1/hosts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    # TODO this is a stopgap. It returns exactly what the UI wants
+    # so we don't have to build associations and make multiple
+    # requests for the /admin/overview/hosts page.
+    # When the client side becomes a little more mature, we should
+    # be make this endpoint a little less terrible and only serve host info
+    host_json = User.hosts.map do |user|
+      {
+        name:                user.name,
+        number_of_teatimes:  user.tea_times.count,
+        last_teatime:        user.tea_times.where('start_time <= ?', Time.now.utc).last,
```
Comment (nickbarnwell) [inline app/controllers/api/v1/hosts_controller.rb]: Can you use `tea_times.past.last` here?

File: app/controllers/api/v1/hosts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        next_teatime:        user.tea_times.where('start_time <= ?', Time.now.utc).last,
```
Comment (nickbarnwell) [inline app/controllers/api/v1/hosts_controller.rb]: Can you `tea_times.future.first` here?

File: app/controllers/api/v1/hosts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        total_attendees:     user.tea_times.map {|t| t.attendances.count}.inject(0, &:+),
```
Comment (nickbarnwell) [inline app/controllers/api/v1/hosts_controller.rb]: This will include wait listed and flaked users in the total attendee count. Is that what you want?

Comment (mehulkar) [inline app/controllers/api/v1/hosts_controller.rb]: Probably a question for @ankitshah811. I just used the same code that was in the Rails view.

**Comment by mehulkar** [comment]
@nickbarnwell addressed your comments except the last one (shows outdated in GH, but wasn't actually addressed)

**Pull request merged**